### ISSUE TITLE
Explicitly protect some tags

### ISSUE THREAD

**Issue by alanfgh** [Tatoeba/tatoeba2#309]
Some users will create bookmarks based on some tags. Based on my experience, the tags we consider important (such as "@change") are unlikely to reach the zero-sentence mark (as nice as that might be) and be deleted automatically. Still, we should add a field (which need not even be editable via the UI in the initial implementation) to protect these tags from deletion. The SQL code would then check this field before attempting to delete the tags.

**Comment by alanfgh** [comment]
This issue should not be assigned to me anymore.

**Comment by jiru** [comment]
If the only problem we’re trying to solve here is to make bookmarks continue to work after the deletion and recreation of a given tag, then we might as well replace the tag ID by its name in the bookmark URL. It’s better than to have to handle more options.

**Comment by trang** [comment]
If tags evolve to have categories (#333) or descriptions, then we will need to explicitly prevent their deletions. Otherwise the following case could happen:
- `@change` is added to some category by some corpus maintainer
- `@change` reaches zero sentences and is deleted
- `@change` is added on a sentence and is re-created, but with no category

Maybe it's not a good idea in general to delete tags automatically when they reach the zero-sentence mark...

**Comment by agrodet** [comment]
In an ideal world, wouldn't it be the case that only utility tags would need to be considered? 

Other tags are very unlikely to be deleted, and if they are it's for a good reason. Talking about the future here, but:   
Remove because incorrect? No need to care.
Merged because synonyms / same thing? No need to care. 
Are there other possibilities?

If only utility tags are concerned, they would belong to the "utility" category. Then, you just have to protect that category, and the job's done.

What do you think?

**Comment by jiru** [comment]
> Merged because synonyms / same thing? No need to care.

In that case, I think the URL of the removed tag should somehow redirect to the URL of the tag it was merged into.

**Comment by alanfgh** [comment]
Yes, I agree that it would be nice to have a special "utility" category, and only tags from that category would be protected.
